### PR TITLE
feat:drop trigger

### LINKED ISSUE #238: Feature: Support drop trigger SQL statement.
**What would you like to be added**:

**Why is this needed**:

https://dev.mysql.com/doc/refman/8.0/en/drop-trigger.html

```
DROP TRIGGER [IF EXISTS] [schema_name.]trigger_name
```

### PR DESCRIPTION
**Pull request by baerwang**
**What this PR does**:

**Which issue(s) this PR fixes**:

Fixes #238 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/executor/redirect.go [hunk 1]
```diff
@@ -288,14 +268,20 @@ func (executor *RedirectExecutor) ExecutorComQuery(ctx *proto.Context) (proto.Re
...
 		}
-
 	}
 
 	executor.doPostFilter(ctx, res)
 
 	return res, warn, err
 }
 
+func isSelectDatabase(ctx *proto.Context, schemaless bool, rt runtime.Runtime) (proto.Result, uint16, error) {
```
Comment (dongzl) [inline pkg/executor/redirect.go:278]: Suggest `isSelectDatabase` not a good name.  Maybe `isXXX` will return `bool` value, but this method logic is execute statement.

Suggest `executeStmt` or other proper name.

**Review by jjeffcaii — COMMENTED** [review]
Please fix

**Review by georgehao — APPROVED** [review]

**Review by dongzl — APPROVED** [review]

File: pkg/runtime/optimize/optimizer.go [hunk 2]
```diff
@@ -1070,3 +1072,36 @@ func (o optimizer) rewriteInsertStatement(ctx context.Context, conn proto.VConn,
...
+	}
+
+	var (
+		shards rule.DatabaseTables
+		err    error
+	)
+
+	if shards, _, err = (*Sharder)(ru).Shard(stmt.Table, nil, args...); err != nil {
+		return nil, errors.Wrap(err, "calculate shards failed")
+	}
```
Comment (jjeffcaii) [inline pkg/runtime/optimize/optimizer.go:1089]: Remove it, no need to shard trigger.

File: pkg/runtime/optimize/optimizer.go [hunk 3]
```diff
@@ -1070,3 +1072,36 @@ func (o optimizer) rewriteInsertStatement(ctx context.Context, conn proto.VConn,
...
+		for _, table := range ru.VTables() {
+			topology := table.Topology()
+			topology.Each(func(dbIdx, tbIdx int) bool {
+				if d, t, ok := topology.Render(dbIdx, tbIdx); ok {
+					shards[d] = append(shards[d], t)
+				}
+				return true
+			})
+		}
+	}
```
Comment (jjeffcaii) [inline pkg/runtime/optimize/optimizer.go:1102]: Using the first vtable is enough.

**Review by jjeffcaii — APPROVED** [review]

**Review by jjeffcaii — APPROVED** [review]
LGTM

**Review by dongzl — APPROVED** [review]

**Pull request merged**